tidy3d/material_library/material_library.py

```diff
@@ -0,0 +1,1645 @@
+"""Holds dispersive models for several commonly used optical materials."""  # pylint: disable=too-many-lines
+from typing import Dict, List
+import pydantic as pd
+from ..components.medium import PoleResidue
+from ..components.base import Tidy3dBaseModel
+from ..log import SetupError
+from .material_reference import material_refs, ReferenceData
+
+
+class VariantItem(Tidy3dBaseModel):
+    """Reference, data_source, and material model for a variant of a material."""
+
+    medium: PoleResidue = pd.Field(
+        ...,
+        title="Material dispersion model",
+        description="A dispersive medium described by the pole-residue pair model.",
+    )
+
+    reference: List[ReferenceData] = pd.Field(
+        None,
+        title="Reference information",
+        description="A list of reference related to this variant model.",
+    )
+
+    data_url: str = pd.Field(
+        None,
+        title="Dispersion data URL",
+        description="The URL to access the dispersion data upon which the material "
+        "model is fitted.",
+    )
+
+
+class MaterialItem(Tidy3dBaseModel):
+    """A material that includes several variants."""
+
+    name: str = pd.Field(..., title="Name", description="Unique name for the medium.")
+    variants: Dict[str, VariantItem] = pd.Field(
+        ...,
+        title="Dictionary of available variants for this material",
+        description="A dictionary of available variants for this material "
+        "that maps from a key to the variant model.",
+    )
+    default: str = pd.Field(
+        ..., title="default variant", description="The default type of variant."
+    )
+
+    @pd.validator("default", always=True)
+    def _default_in_variants(cls, val, values):
+        """Make sure the default variant is already included in the ``variants``."""
+        if not val in values["variants"].keys():
```

Comment:
no need for `.keys()` here I think, just `if not val in values['variants']` already searches through keys